### ISSUE TITLE
Avahi ignoring network interface with dhcpcd 9

### ISSUE THREAD

**Issue by slokhorst** [lathiat/avahi#279]
Since a little while, avahi-daemon does not start on my main network interface. I have no idea why this is happening. Avahi used to work fine before.

I'm using avahi 0.8+15+ge8a3dd0-1 on Arch Linux.

I have two network interfaces on my system. The default loopback (`lo`) and my actual network adapter (`enp3s0`):
```
$ ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host 
       valid_lft forever preferred_lft forever
2: enp3s0: <BROADCAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
    link/ether d4:3d:7e:bc:9e:fc brd ff:ff:ff:ff:ff:ff
    inet 192.168.178.66/24 brd 192.168.178.255 scope global dynamic noprefixroute enp3s0
       valid_lft 3368sec preferred_lft 2918sec
    inet6 2a02:a212:9281:2480:d63d:7eff:febc:9efc/64 scope global dynamic mngtmpaddr noprefixroute 
       valid_lft 1079496sec preferred_lft 474696sec
    inet6 fe80::d63d:7eff:febc:9efc/64 scope link 
       valid_lft forever preferred_lft forever
```

There are no other mDNS services running:
```
$ sudo lsof -i -P -n | grep 5353
(no results)
```

If I run avahi-daemon, it only picks up `lo` and seems to ignore `enp3s0`:
```
$ sudo avahi-daemon
Process 3248 died: No such process; trying to remove PID file. (/run/avahi-daemon//pid)
Found user 'avahi' (UID 84) and group 'avahi' (GID 84).
Successfully dropped root privileges.
avahi-daemon 0.8 starting up.
Successfully called chroot().
Successfully dropped remaining capabilities.
No service file found in /etc/avahi/services.
Joining mDNS multicast group on interface lo.IPv6 with address ::1.
New relevant interface lo.IPv6 for mDNS.
Joining mDNS multicast group on interface lo.IPv4 with address 127.0.0.1.
New relevant interface lo.IPv4 for mDNS.
Network interface enumeration completed.
Registering new address record for ::1 on lo.*.
Registering new address record for 127.0.0.1 on lo.IPv4.
Server startup complete. Host name is SebbyTron.local. Local service cookie is 2554207456.
^CGot SIGINT, quitting.
Leaving mDNS multicast group on interface lo.IPv6 with address ::1.
Leaving mDNS multicast group on interface lo.IPv4 with address 127.0.0.1.
avahi-daemon 0.8 exiting.
```

My `avahi-daemon.conf` is very basic:
```
$ cat /etc/avahi/avahi-daemon.conf
[server]
use-ipv4=yes
use-ipv6=yes
```

**Comment by slokhorst** [comment]
This issue seems to be caused by dhcpcd 9. After I rolled back to dhcpcd 8.1.7, avahi again starts on `enp3s0`:

```
$ sudo avahi-daemon
Found user 'avahi' (UID 84) and group 'avahi' (GID 84).
Successfully dropped root privileges.
avahi-daemon 0.8 starting up.
Successfully called chroot().
Successfully dropped remaining capabilities.
No service file found in /etc/avahi/services.
Joining mDNS multicast group on interface enp3s0.IPv6 with address 2a02:a212:9281:2480:d63d:7eff:febc:9efc.
New relevant interface enp3s0.IPv6 for mDNS.
Joining mDNS multicast group on interface enp3s0.IPv4 with address 192.168.178.66.
New relevant interface enp3s0.IPv4 for mDNS.
Joining mDNS multicast group on interface lo.IPv6 with address ::1.
New relevant interface lo.IPv6 for mDNS.
Joining mDNS multicast group on interface lo.IPv4 with address 127.0.0.1.
New relevant interface lo.IPv4 for mDNS.
Network interface enumeration completed.
Registering new address record for 2a02:a212:9281:2480:d63d:7eff:febc:9efc on enp3s0.*.
Registering new address record for 192.168.178.66 on enp3s0.IPv4.
Registering new address record for ::1 on lo.*.
Registering new address record for 127.0.0.1 on lo.IPv4.
Server startup complete. Host name is SebbyTron.local. Local service cookie is 1531598634.
```

**Comment by slokhorst** [comment]
Also posted on the dhcpcd mailing list: https://roy.marples.name/archives/dhcpcd-discuss/0002927.html

**Comment by rsmarples** [comment]
I'm upstream for dhcpcd, but there should be no reason why dhcpcd-9 should stop anything from working. I don't use avahi, so unsure how to help other than to request working and non working `ip a` output - it's not clear from your post if it's non working or working or if there is any difference.

**Comment by slokhorst** [comment]
Thanks for your comment @rsmarples. I hadn't thought of comparing those, but I think you are on the right track!

```
# `ip addr` when using dhcpcd 8:
enp3s0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000

# `ip addr` when using dhcpcd 9:
enp3s0: <BROADCAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UP group default qlen 1000
```

(the rest of the lines is the same)

Notably, `MULTICAST` has disappeared! Any idea what might be causing this? Is this an option I can set in dhcpcd?

**Comment by rsmarples** [comment]
GAAHHHH. So it is a dhcpcd fault!!
Fixed in rsmarples/dhcpcd@097c471

Can you test that please?

**Comment by slokhorst** [comment]
Yep, that fixes it! Thanks for the fast fix @rsmarples !
